### PR TITLE
[Feat] Add Gradient Boosted CIF

### PR DESCRIPTION
**Pull request by Vincent-Maladiere**
- I could reproduce the results of the [tutorial part 1](https://github.com/soda-inria/survival-analysis-benchmark/blob/main/notebooks/tutorial_part_1.ipynb) with this updated version of the model
- Need some feedback about the `y` input of fit. For now, it accepts record arrays, dataframes and dict provided that the dtypes/columns/keys names are `["event", "duration"]`, and converts everything to a recarray internally.

**TODO**:
- [ ] add doc
- [ ] add tests
- [ ] add exemples

cc @ogrisel

### CODE REVIEW

**Comment by ogrisel** [comment]
To ease the review and ensure proper testing, let's split into more focussed PRs:

- one for the brier score metrics and its doc + tests
- one for the GB CIF model + its doc + test and examples

The second PR can be a branch based on top of the branch of the first PR.

For the name of the CIF estimator, maybe we could go for `GradientBoostingIncidence` instead of `GradientBoostedCIF` to avoid acronyms.

File: hazardous/survival_mixin.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import numpy as np
+
+
+class SurvivalMixin:
+    def predict_cumulative_hazard_function(self, X_test, times):
+        survival_probs = self.predict_survival_function(X_test, times)
+        cumulative_hazards = -np.log(survival_probs + 1e-8)
+        return cumulative_hazards
```
Comment (ogrisel) [inline hazardous/survival_mixin.py:8]: I would rather not introduce a function that method that derives a cumulative hazard from a survival function estimate as this cannot generalize to the competing risks setting. Let's remove this mixin.

Comment (Vincent-Maladiere) [reply hazardous/survival_mixin.py:8]: yes indeed, let's remove it

**Comment by Vincent-Maladiere** [comment]
Closing in favor of #10 and #11

**Pull request closed without merging**